src/classes/RD_RecurringDonations.cls

```diff
@@ -151,103 +151,115 @@ public class RD_RecurringDonations {
-        //and the paid ammounts and paid installments 
-        for (sobject obj : [select count(id) oppcount, MIN(CloseDate) MinDate, MAX(CloseDate) MaxDate, SUM(Amount) Total,
-                                  isClosed, npe03__Recurring_Donation__c rdid, isWon
-                                  from Opportunity where npe03__Recurring_Donation__r.npe03__Open_Ended_Status__c <> :closedLabel 
-                                  and npe03__Recurring_Donation__c IN :rdMap.keySet()
-                                  group by rollup(npe03__Recurring_Donation__c, isClosed, isWon)]) {
-            id rdid = (id)obj.get('rdid'); 
-            
-            boolean isClosed = (boolean)obj.get('isClosed');
-            boolean isWon = (boolean)obj.get('isWon');            
-            
-            if (rdid != null) {
-                npe03__Recurring_Donation__c rd;        
-                //if we already have values for this RD, get it from our rdid map, otherwise just grab
-                //the one we're already working on 
-                if (!updateMap.containsKey(rdid)) {
-                    rd = rdMap.get(rdid);
-
-                    //initialize fields to null that might not get set
-                    rd.npe03__Next_Payment_Date__c = null;
-                    rd.npe03__Last_Payment_Date__c = null;
-                    rd.npe03__Paid_Amount__c = null;     
-                    rd.npe03__Total_Paid_Installments__c = null;
-                    updateMap.put(rdid, rd);
-                } else {
-                    rd = updateMap.get(rdid);
-                }
-                
-                //if this is the summary row for this rd only update installments for open ended opportunities
-                if (isClosed == null && isWon == null && rd.npe03__open_ended_status__c == system.label.npe03.RecurringDonationOpenStatus) {
-                    rd.npe03__Installments__c = (integer)obj.get('oppcount'); 
-                    updateMap.put(rdid, rd);
-                }
-                //if this is the open set of opps
-                else if (isClosed == false && isWon == null) {
-                    rd.npe03__Next_Payment_Date__c = (date)obj.get('MinDate');
-                    updateMap.put(rdid, rd);
-                }
-                //if this is the closed set of opps    
-                else if (isClosed == true && isWon == true) {
-                    rd.npe03__Last_Payment_Date__c = (date)obj.get('MaxDate');
 
-                    //convert rollup total to RD's currency
-                    if (RLLP_OppRollup_UTIL.isMultiCurrency()) {
-                        rd.npe03__Paid_Amount__c = UTIL_CurrencyConversion.convertFromCorporateUsingStandardRates((String) rd.get(RDCurrencyField), (Decimal) obj.get('Total'));
+        Boolean CRLPEnabled = CRLP_Rollup_SVC.isCustomizableRollupEngineEnabled;
+
+        // only use this hard-coded RD rollup code if Customizable Rollups aren't enabled
+        if (!CRLPEnabled) {
```

Comment:
@bethbrains After running through a test on this in a scratch org and comparing the new functionality to the old functionality, the only difference I found is that the "Next Payment Date" is not automatically populated after creating a new RD when Customizable Rollups are enabled. I suspect this is because the new rollups are not running after the installment Opps are created. I think this is because there is no associated `else` with this condition. Basically, it's calculating the values if CRLP is disabled, but nothing happens when CRLP is enabled.

Reply:
The trigger does pick it up as designed, it's just LEX caching. I've added this to the tests to verify, and tested in the UI by turning off the persistent caching setting and refreshing after RD creation. Arrrrrgh LEX caching...